### ISSUE TITLE
simuPOP does not play well with feather

### ISSUE THREAD

**Issue by BoPeng** [BoPeng/simuPOP#7]
https://github.com/wesm/feather/issues/227

Hello,

This might be a non-issue, but I wanted to bring it to your attention in the case there are other conflicts.

I found out that I had an issue with import order of feather (here's a working example) and another module called simuPOP. This was only an issue on Ubuntu (14.04 and 16.04), it worked fine on OSX.

If feather was imported before simuPOP, an error from simuPOP was raised. When feather was imported after simuPOP, it was no issue.

Here's a MWE with conda:

```
conda update --all 
conda install python=3.4 
conda install -c https://conda.binstar.org/bpeng simuPOP
conda install numpy
pip install feather-format

git clone https://github.com/zkamvar/simupop-example
cd simupop-example/testpopp

# fails
python -c 'import feather;from simuPOP import *;loadPopulation("seed_0_sex_0.0000_gen_00000_rep_00.pop")'

# works
python -c 'from simuPOP import *;import feather;loadPopulation("seed_0_sex_0.0000_gen_00000_rep_00.pop")'
```

**Comment by BoPeng** [comment]
I cannot reproduce this bug with latest version of simuPOP and mac OSX.

**Comment by zkamvar** [comment]
As noted above, it has previously worked with OSX and simupop:

```sh
08:04:00~/Documents/simupop-example/testpopp (master)> python -c 'import feather;from simuPOP import *;loadPopulation("seed_0_sex_0.0000_gen_00000_rep_00.pop")'
simuPOP Version 1.1.7 : Copyright (c) 2004-2016 Bo Peng
Revision 5000 (Jan 21 2016) for Python 3.4.4 (64bit, 0thread)
Random Number Generator is set to mt19937 with random seed 0xa34142b7418d0214.
This is the standard short allele version with 256 maximum allelic states.
For more information, please visit http://simupop.sourceforge.net,
or email simupop-list@lists.sourceforge.net (subscription required).
08:04:05~/Documents/simupop-example/testpopp (master)>
```

I'll try the latest version on Ubuntu in a couple of days when I have access to a machine.

**Comment by BoPeng** [comment]
OK. I will also try it on a linux machine. I would like to release the next version of simuPOP and need to clear this bug before I do that.

**Comment by zkamvar** [comment]
Hi, I just did a test on a 32 bit version of Ubuntu 16.04.1 and can confirm that I can no longer reproduce the bug. It works ^_^

**Comment by BoPeng** [comment]
It is possible that binstar version uses different version of runtime libraries than the local machine, but I remember I compiled from source and confirmed the problem. Anyway, I will test on the platforms that I compile binary for the next release and see if I can reproduce it.

**Comment by zkamvar** [comment]
To be clear: I'm not finding this bug anymore. Does this mean you fixed it in https://github.com/BoPeng/simuPOP/commit/435b6687700055b0ab80b7d09bcecd029652a970?

I used a brand new distribution of Ubuntu and installed everything as shown above (except simuPOP, which I compiled from source).

**Comment by BoPeng** [comment]
The symptoms are similar but that seg-fault was caused by incorrect release of a cross-layer (python <-> c/c++) function, but this one was caused by a simple load of population, which should not trigger any such function.

**Comment by BoPeng** [comment]
A new RHEL machine, anaconda python 3.4,

```
python -c 'import feather;from simuPOP import *;loadPopulation("seed_0_sex_0.0000_gen_00000_rep_00.pop")'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/bpeng1/anaconda3/lib/python3.4/site-packages/simuPOP/__init__.py", line 323, in <module>
    from simuPOP.simuPOP_std import *
  File "/home/bpeng1/anaconda3/lib/python3.4/site-packages/simuPOP/simuPOP_std.py", line 32, in <module>
    _simuPOP_std = swig_import_helper()
  File "/home/bpeng1/anaconda3/lib/python3.4/site-packages/simuPOP/simuPOP_std.py", line 28, in swig_import_helper
    _mod = imp.load_module('_simuPOP_std', fp, pathname, description)
  File "/home/bpeng1/anaconda3/lib/python3.4/imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
ImportError: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /home/bpeng1/anaconda3/lib/python3.4/site-packages/simuPOP/_simuPOP_std.cpython-34m.so)
```

Need to compile from source.

**Comment by ashander** [comment]
Importing `feather` first produces a `ValueError` on ubuntu `Linux version 4.8.0-32-generic (buildd@lcy01-34) (gcc version 6.2.0 20161005 (Ubuntu 6.2.0-5ubuntu12) ) #34-Ubuntu SMP Tue Dec 13 14:30:43 UTC 2016`

```sh
python -c 'import feather;from simuPOP import *;loadPopulation("seed_0_sex_0.0000_gen_00000_rep_00.pop")'
simuPOP Version 1.1.8svn : Copyright (c) 2004-2016 Bo Peng
Revision 5003 (Jan 13 2017) for Python 3.6.0 (64bit, 1thread)
Random Number Generator is set to mt19937 with random seed 0x979dd93eddea6933.
This is the standard short allele version with 256 maximum allelic states.
For more information, please visit http://simupop.sourceforge.net,
or email simupop-list@lists.sourceforge.net (subscription required).
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/jaime/miniconda3/envs/simuPOP/lib/python3.6/site-packages/simuPOP-1.1.8svn-py3.6-linux-x86_64.egg/simuPOP/simuPOP_std.py", line 4773, in loadPopulation
    return _simuPOP_std.loadPopulation(file)
ValueError: Failed to load Population seed_0_sex_0.0000_gen_00000_rep_00.pop.

```

python environment
```
 $ conda list
# packages in environment at /home/jaime/miniconda3/envs/simuPOP:
#
Cython                    0.25.2                    <pip>
decorator                 4.0.10                   py36_1  
feather-format            0.3.1                     <pip>
ipython                   5.1.0                    py36_0  
ipython_genutils          0.1.0                    py36_0  
numpy                     1.11.3                    <pip>
openssl                   1.0.2j                        0  
pandas                    0.19.2                    <pip>
path.py                   9.0.1                    py36_0  
pexpect                   4.0.1                    py36_0  
pickleshare               0.7.4                    py36_0  
pip                       9.0.1                    py36_1  
prompt_toolkit            1.0.9                    py36_0  
ptyprocess                0.5.1                    py36_0  
pygments                  2.1.3                    py36_0  
python                    3.6.0                         0  
python-dateutil           2.6.0                     <pip>
pytz                      2016.10                   <pip>
readline                  6.2                           2  
setuptools                27.2.0                   py36_0  
simplegeneric             0.8.1                    py36_1  
simuPOP                   1.1.8svn                  <pip>
six                       1.10.0                   py36_0  
sqlite                    3.13.0                        0  
tk                        8.5.18                        0  
traitlets                 4.3.1                    py36_0  
wcwidth                   0.1.7                    py36_0  
wheel                     0.29.0                   py36_0  
xz                        5.2.2                         1  
zlib                      1.2.8                         3  
```

**Comment by BoPeng** [comment]
Does the other command (load simuPOP first) pass?

**Comment by ashander** [comment]
Yes. No problem when loading first:

```
$ python -c 'from simuPOP import *;import feather; loadPopulation("seed_0_sex_0.0000_gen_00000_rep_00.pop")'
simuPOP Version 1.1.8svn : Copyright (c) 2004-2016 Bo Peng
Revision 5003 (Jan 13 2017) for Python 3.6.0 (64bit, 1thread)
Random Number Generator is set to mt19937 with random seed 0x2cbcae5503ed3406.
This is the standard short allele version with 256 maximum allelic states.
For more information, please visit http://simupop.sourceforge.net,
or email simupop-list@lists.sourceforge.net (subscription required).
```

**Comment by BoPeng** [comment]
Python 3.4.4, simuPOP 1.1.7, Ubuntu 16.4 LTS, cannot reproduce. Checking 1.1.8svn now.

```
bpeng@BenPC:~/simuPOP/test/simupop-example/testpopp$$ python -c 'import feather;from simuPOP import *;loadPopulation("seed_0_sex_0.0000_gen_00000_rep_00.pop")'
simuPOP Version 1.1.7 : Copyright (c) 2004-2016 Bo Peng
Revision 5000 (Jan 21 2016) for Python 3.4.4 (64bit, 1thread)
Random Number Generator is set to mt19937 with random seed 0x6c86c584f26bab78.
This is the standard short allele version with 256 maximum allelic states.
For more information, please visit http://simupop.sourceforge.net,
or email simupop-list@lists.sourceforge.net (subscription required).

bpeng@BenPC:~/simuPOP/test/simupop-example/testpopp$ python -c 'from simuPOP import *;import feather;loadPopulation("seed_0_sex_0.0000_gen_00000_rep_00.pop")'
simuPOP Version 1.1.7 : Copyright (c) 2004-2016 Bo Peng
Revision 5000 (Jan 21 2016) for Python 3.4.4 (64bit, 1thread)
Random Number Generator is set to mt19937 with random seed 0x849a16fc5cf542a.
This is the standard short allele version with 256 maximum allelic states.
For more information, please visit http://simupop.sourceforge.net,
or email simupop-list@lists.sourceforge.net (subscription required).

bpeng@BenPC:~/simuPOP/test/simupop-example/testpopp$ conda list
# packages in environment at /home/bpeng/anaconda3:
#
bokeh                     0.12.4                   py35_0  
boto                      2.45.0                   py35_0  
bottleneck                1.2.0               np111py35_0  
cffi                      1.9.1                    py34_0  
chardet                   2.3.0                    py35_0  
chest                     0.2.3                    py35_0  
click                     6.6                      py35_0  
cloudpickle               0.2.2                    py35_0  
clyent                    1.2.2                    py35_0  
colorama                  0.3.7                    py35_0  
conda                     4.3.4                    py34_0  
conda-build               2.1.1                    py35_0  
conda-env                 2.6.0                         0  
conda-verify              2.0.0                    py35_0  
configobj                 5.0.6                    py35_0  
contextlib2               0.5.4                    py35_0  
cryptography              1.7.1                    py34_0  
curl                      7.49.0                        1  
cycler                    0.10.0                   py35_0  
Cython                    0.25.2                    <pip>
cython                    0.25.2                   py35_0  
cytoolz                   0.8.2                    py35_0  
dask                      0.13.0                   py35_0  
datashape                 0.5.4                    py35_0  
dbus                      1.10.10                       0  
decorator                 4.0.10                   py35_1  
dill                      0.2.5                    py35_0  
docutils                  0.12                     py35_2  
dynd-python               0.7.2                    py35_0  
entrypoints               0.2.2                    py35_0  
et_xmlfile                1.0.1                    py35_0  
expat                     2.1.0                         0  
fastcache                 1.0.2                    py35_1  
feather-format            0.3.1                     <pip>
filelock                  2.0.7                    py35_0  
flask                     0.12                     py35_0  
flask-cors                2.1.2                    py35_0  
fontconfig                2.12.1                        0  
freetype                  2.5.5                         1  
get_terminal_size         1.0.0                    py35_0  
gevent                    1.2.0                    py35_0  
glib                      2.50.2                        0  
greenlet                  0.4.11                   py35_0  
gst-plugins-base          1.8.0                         0  
gstreamer                 1.8.0                         0  
h5py                      2.6.0               np111py35_2  
hdf5                      1.8.17                        1  
heapdict                  1.0.0                    py35_1  
icu                       54.1                          0  
idna                      2.1                      py34_0  
imagesize                 0.7.1                    py35_0  
ipykernel                 4.5.2                    py35_0  
ipython                   5.1.0                    py35_0  
ipython_genutils          0.1.0                    py35_0  
ipywidgets                5.2.2                    py35_0  
itsdangerous              0.24                     py35_0  
jbig                      2.1                           0  
jdcal                     1.3                      py35_0  
jedi                      0.9.0                    py35_1  
jinja2                    2.8.1                    py35_0  
jpeg                      8d                            2  
jsonschema                2.5.1                    py35_0  
jupyter                   1.0.0                    py35_3  
jupyter_client            4.4.0                    py35_0  
jupyter_console           5.0.0                    py35_0  
jupyter_core              4.2.1                    py35_0  
lazy-object-proxy         1.2.1                    py35_0  
libdynd                   0.7.2                         0  
libffi                    3.2.1                         1  
libgcc                    5.2.0                         0  
libgfortran               3.0.0                         1  
libiconv                  1.14                          0  
libpng                    1.6.27                        0  
libsodium                 1.0.10                        0  
libtiff                   4.0.6                         2  
libxcb                    1.12                          1  
libxml2                   2.9.4                         0  
libxslt                   1.1.29                        0  
llvmlite                  0.15.0                   py35_0  
locket                    0.2.0                    py35_1  
lxml                      3.7.2                    py35_0  
markupsafe                0.23                     py35_2  
matplotlib                1.5.3               np111py35_1  
mistune                   0.7.3                    py35_0  
mkl                       2017.0.1                      0  
mkl-service               1.1.2                    py34_3  
mpmath                    0.19                     py35_1  
multipledispatch          0.4.9                    py35_0  
nb_anacondacloud          1.2.0                    py35_0  
nb_conda                  2.0.0                    py35_0  
nb_conda_kernels          2.0.0                    py35_0  
nbconvert                 4.2.0                    py35_0  
nbformat                  4.2.0                    py35_0  
nbpresent                 3.0.2                    py35_0  
networkx                  1.11                     py35_0  
nltk                      3.2.2                    py35_0  
nose                      1.3.7                    py34_1  
notebook                  4.3.1                    py35_0  
numba                     0.30.1              np111py35_0  
numexpr                   2.6.1               np111py34_2  
numpy                     1.11.3                   py34_0  
odo                       0.5.0                    py35_1  
openpyxl                  2.4.0                    py35_0  
openssl                   1.0.2j                        0  
pandas                    0.19.2                    <pip>
pandas                    0.19.2              np111py35_1  
partd                     0.3.7                    py35_0  
patchelf                  0.9                           0  
path.py                   9.0.1                    py35_0  
pathlib2                  2.1.0                    py35_0  
patsy                     0.4.1                    py35_0  
pcre                      8.39                          1  
pep8                      1.7.0                    py35_0  
pexpect                   4.0.1                    py35_0  
pickleshare               0.7.4                    py35_0  
pillow                    3.4.2                    py35_0  
pip                       9.0.1                    py34_1  
pkginfo                   1.4.1                    py35_0  
ply                       3.9                      py35_0  
prompt_toolkit            1.0.9                    py35_0  
psutil                    5.0.1                    py35_0  
ptyprocess                0.5.1                    py35_0  
py                        1.4.31                   py35_0  
pyasn1                    0.1.9                    py34_0  
pycosat                   0.6.1                    py34_1  
pycparser                 2.17                     py34_0  
pycrypto                  2.6.1                    py35_4  
pycurl                    7.43.0                   py35_0  
pyflakes                  1.4.0                    py35_0  
pygments                  2.1.3                    py35_0  
pylint                    1.5.4                    py35_1  
pyopenssl                 16.2.0                   py34_0  
pyparsing                 2.1.4                    py35_0  
pyqt                      5.6.0                    py35_1  
pytables                  3.3.0               np111py35_0  
pytest                    3.0.5                    py35_0  
python                    3.4.5                         0  
python-dateutil           2.6.0                     <pip>
python-dateutil           2.6.0                    py35_0  
pytz                      2016.10                  py35_0  
pytz                      2016.10                   <pip>
pyyaml                    3.12                     py34_0  
pyzmq                     16.0.2                   py35_0  
qt                        5.6.2                         2  
qtawesome                 0.4.1                    py35_0  
qtconsole                 4.2.1                    py35_1  
qtpy                      1.1.2                    py35_0  
readline                  6.2                           2  
redis                     3.2.0                         0  
redis-py                  2.10.5                   py35_0  
requests                  2.12.4                   py34_0  
rope                      0.9.4                    py35_1  
ruamel_yaml               0.11.14                  py34_1  
scikit-image              0.12.3              np111py35_1  
scikit-learn              0.18.1              np111py34_1  
scipy                     0.18.1              np111py34_1  
setuptools                27.2.0                   py34_0  
simplegeneric             0.8.1                    py35_1  
simupop                   1.1.7                    py34_0    https://conda.binstar.org/bpeng
singledispatch            3.4.0.3                  py35_0  
sip                       4.18                     py35_0  
six                       1.10.0                   py34_0  
snowballstemmer           1.2.1                    py35_0  
sockjs-tornado            1.0.3                    py35_0  
sphinx                    1.5.1                    py35_0  
sphinx_rtd_theme          0.1.9                    py35_0  
spyder                    3.0.2                    py35_0  
sqlalchemy                1.1.4                    py35_0  
sqlite                    3.13.0                        0  
statsmodels               0.6.1               np111py35_1  
sympy                     1.0                      py35_0  
terminado                 0.6                      py35_0  
tk                        8.5.18                        0  
toolz                     0.8.2                    py35_0  
tornado                   4.4.2                    py35_0  
traitlets                 4.3.1                    py35_0  
unicodecsv                0.14.1                   py35_0  
wcwidth                   0.1.7                    py35_0  
werkzeug                  0.11.13                  py35_0  
wheel                     0.29.0                   py34_0  
widgetsnbextension        1.2.6                    py35_0  
wrapt                     1.10.8                   py35_0  
xlrd                      1.0.0                    py35_0  
xlsxwriter                0.9.6                    py35_0  
xlwt                      1.1.2                    py35_0  
xz                        5.2.2                         1  
yaml                      0.1.6                         0  
zeromq                    4.1.5                         0  
zlib                      1.2.8                         3  
```

**Comment by BoPeng** [comment]
I created a Ubuntu 16.04 vm, installed anaconda python 3.6.

1. Installed `feather-format` from `pip` failed so I installed it from `conda`.
2. Installed `simuPOP` from the latest source. simuPOP compiled fine but could not be imported. The error message pointed to some GOMP (openMP library) problem. It appeared that simuPOP was compiled with system gomp, but linked to anaconda gomp. I removed simuPOP, installed conda gcc (conda install gcc), recompiled simuPOP and simuPOP imported fine.

Then I could **not** reproduce the bug.

I have spent quite some time on this issue and I do not think it is a wise to continue. I  will keep this ticket open in case someone else encounters this problem.
